### PR TITLE
Add DRF Spectacular with Swagger and Redoc for API Documentation

### PR DESCRIPTION
**Pull request by okt-limonikas**
## **Description**
This PR introduces DRF Spectacular to generate OpenAPI documentation for our API and integrates Swagger UI and Redoc as interactive documentation tools.  
This is an initial step toward comprehensive API documentation.
With this integration, clients will now be able to generate code in various programming languages to interact with the API, leveraging the OpenAPI schema.

## **Key Changes**
- Added **DRF Spectacular** for OpenAPI schema generation.
- Configured **Swagger UI** and **Redoc** for visualising and exploring the API.
- Integrated schema URLs:
  - Swagger: `/schema/swagger-ui/`
  - Redoc: `/schema/redoc/`
  - Schema: `/schema/swagger.yaml`
- Adjusted viewsets to ensure schema generation works without errors (e.g., adding queryset handling and filter overrides for unsupported fields like `JSONField`).

## **Next Steps**
- New endpoints should include detailed API documentation using DRF Spectacular decorators such as:
  - `@extend_schema`
  - `@extend_schema_field`
- Incrementally document existing endpoints to ensure consistency and clarity across the API.

## **How to Test**
1. Ensure you installed new dependencies.
2. Ensure you have updated `settings.py` with drf-spectacular settings added
3. Navigate to:
   - **Swagger UI**: `/api/v2/schema/swagger-ui/`
   - **Redoc**: `/api/v2/schema/redoc`
4. Verify that all currently documented endpoints are correctly listed and functional in both Swagger and Redoc.

### CODE REVIEW

File: bublik/urls.py [hunk 1]
```diff
@@ -59,6 +60,9 @@
     # V2
     re_path(r'^v2/$', api_v2.render_react, name='ui-v2-index'),
     re_path(r'^v2/(?:.*)/?$', api_v2.render_react, name='ui-v2-routes'),
+    path('api/v2/schema/swagger.yaml', SpectacularAPIView.as_view(), name='schema'),
+    path('api/v2/schema/swagger-ui', SpectacularSwaggerView.as_view(url_name='schema'), name='swagger-ui'),
```
Comment (ol-nata) [inline bublik/urls.py:64]: 1. The required slash is missing at the end of the path.
2. The line is too long (will be fixed as a result of applying the linter).

File: bublik/urls.py [hunk 2]
```diff
@@ -59,6 +60,9 @@
...
+    path('api/v2/schema/redoc', SpectacularRedocView.as_view(url_name='schema'), name='redoc'),
```
Comment (ol-nata) [inline bublik/urls.py:65]: The required slash is missing at the end of the path.

File: bublik/urls.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
 from django.urls import include, path, re_path
 from django.views.decorators.cache import cache_page
 from rest_framework.routers import DefaultRouter
+from drf_spectacular.views import SpectacularAPIView, SpectacularRedocView, SpectacularSwaggerView
```
Comment (ol-nata) [inline bublik/urls.py:9]: The order of the import blocks is incorrect and the line is too long (will be fixed as a result of applying the linter).

File: bublik/interfaces/api_v2/config.py [hunk 4]
```diff
@@ -18,13 +18,31 @@
...
+from django.contrib.postgres.fields import JSONField
+
 
 logger = logging.getLogger('')
 
 
+class ConfigFilterSet(filters.FilterSet):
+    class Meta:
+        model = Config
+        fields = ['type', 'name', 'is_active', 'version']
```
Comment (ol-nata) [inline bublik/interfaces/api_v2/config.py:31]: Mutable class attributes should be annotated with `typing.ClassVar`.

File: bublik/interfaces/api_v2/config.py [hunk 5]
```diff
@@ -18,13 +18,31 @@
...
+        filter_overrides = {
```
Comment (ol-nata) [inline bublik/interfaces/api_v2/config.py:32]: Mutable class attributes should be annotated with `typing.ClassVar`.

File: bublik/interfaces/api_v2/config.py [hunk 6]
```diff
@@ -18,13 +18,31 @@
 from bublik.data.models import Config, ConfigTypes, GlobalConfigNames
 from bublik.data.serializers import ConfigSerializer
 
+from django_filters import rest_framework as filters
```
Comment (ol-nata) [inline bublik/interfaces/api_v2/config.py:21]: The order of the import blocks is incorrect (will be fixed as a result of applying the linter).

File: bublik/interfaces/api_v2/config.py [hunk 7]
```diff
@@ -18,13 +18,31 @@
...
+class ConfigFilterSet(filters.FilterSet):
+    class Meta:
+        model = Config
+        fields = ['type', 'name', 'is_active', 'version']
+        filter_overrides = {
+            JSONField: {
+                'filter_class': filters.CharFilter,
+                'extra': lambda f: {
+                    'lookup_expr': 'icontains',
+                }
```
Comment (ol-nata) [inline bublik/interfaces/api_v2/config.py:37]: The trailing comma is missing (will be fixed as a result of applying the linter).

**Review by ol-nata — CHANGES_REQUESTED** [review]
It is necessary to apply the linter to all modified files (see https://github.com/ts-factory/bublik?tab=readme-ov-file#development).

File: requirements.txt [hunk 8]
```diff
@@ -104,3 +104,5 @@ XStatic-Patternfly-Bootstrap-Treeview==2.1.3.2
 yamale==1.8.1
 yapf==0.31.0
 zipp==2.0.0
+drf-spectacular==0.27.2
+drf-spectacular-sidecar==2024.11.1
```
Comment (ol-nata) [inline requirements.txt:108]: Incorrect order (we use alphabetical).

**Review by ol-nata — CHANGES_REQUESTED** [review]

File: requirements.txt [hunk 9]
```diff
@@ -19,7 +19,6 @@ coverage==5.3
 cryptography==2.5
 deepdiff==6.2.3
 Dickens==2.0.0
-Django==3.2.12
```
Comment (arybchik) [inline requirements.txt:22]: I don't understand why previous sorting is bad

File: requirements.txt [hunk 10]
```diff
@@ -33,6 +33,8 @@ django-static-jquery==2.1.4
 Django==3.2.12
 djangorestframework-simplejwt
 djangorestframework==3.13.1
+drf-spectacular-sidecar==2024.11.1
```
Comment (arybchik) [inline requirements.txt:36]: It is unclear why we need these dependencies. Changeset description does not shed any light on it.

File: bublik/urls.py [hunk 11]
```diff
@@ -5,6 +5,11 @@
 from django.contrib import admin
```
Comment (arybchik) [inline bublik/urls.py:5]: A bit different style of referencing to issue is used in the repository. Please, follow existing style.

File: templates/settings.py.template [hunk 12]
```diff
@@ -59,6 +59,8 @@ INSTALLED_APPS = [
     'rest_framework',
```
Comment (arybchik) [inline templates/settings.py.template:59]: Please, follow already used style of referencing to issue in the changeset description.

File: bublik/interfaces/api_v2/config.py [hunk 13]
```diff
@@ -188,8 +188,8 @@ def destroy(self, request, *args, **kwargs):
         return super().destroy(request, *args, **kwargs)
 
     @auth_required(as_admin=True)
-    @action(detail=False, methods=['get'])
-    def schema(self, request, *args, **kwargs):
+    @action(detail=False, methods=['get'], url_path='schema')
+    def get_schema(self, request, *args, **kwargs):
```
Comment (arybchik) [inline bublik/interfaces/api_v2/config.py:209]: It is unclear why happens before the patch. Is everything broken? If so, it should be squashed into previous changeset where drf_spectacular is added.

**Review by arybchik — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 9 above)
Comment (okt-limonikas) [reply requirements.txt:22]: Updated commit description:
```
Issues with current sorting:
    - Sorting is inconsistent due to case sensitivity.
      For example, `Django` is listed below `django-*` packages,
      while it should precede them in a case-insensitive alphabetical order
    - Proper alphabetical sorting requires ignoring case differences,
      ensuring Django appears before `django-*`
```
      
There is currently no documented standard for sorting `requirements.txt`, which leads to inconsistencies. To ensure uniformity, we should use a single tool or command
Maybe I missed on how to sort `requirements.txt`

Probably should create issue if we don't have a consistent way to sort `requirements.txt`?

File: requirements.txt (hunk 10 above)
Comment (okt-limonikas) [reply requirements.txt:36]: Added description

File: bublik/interfaces/api_v2/config.py (hunk 13 above)
Comment (okt-limonikas) [reply bublik/interfaces/api_v2/config.py:209]: The conflict did not break the application but prevented correct schema generation.
No clear error message was raised, making the issue difficult to debug.
I would prefer this change to be explicit for clarity and traceability

Clarified commit description

File: bublik/interfaces/api_v2/config.py [hunk 14]
```diff
@@ -188,8 +188,8 @@ def destroy(self, request, *args, **kwargs):
         return super().destroy(request, *args, **kwargs)
```
Comment (arybchik) [inline bublik/interfaces/api_v2/config.py:205]: Don't use past tense in the summary and description.

File: bublik/interfaces/api_v2/config.py [hunk 15]
```diff
@@ -3,10 +3,12 @@
```
Comment (arybchik) [inline bublik/interfaces/api_v2/config.py:3]: Don't use past tense in the summary.

Also summary is full is technical details which are very-very specific and reader may not know it.
Just say why the change is needed in the summary. Which problems is solved? Ideally without method names etc.
Technical details are allowed in the description, but not summary.

**Review by arybchik — CHANGES_REQUESTED** [review]

File: bublik/interfaces/api_v2/config.py (hunk 15 above)
Comment (okt-limonikas) [reply bublik/interfaces/api_v2/config.py:3]: Improved summary

File: bublik/interfaces/api_v2/config.py (hunk 14 above)
Comment (okt-limonikas) [reply bublik/interfaces/api_v2/config.py:205]: Fixed

File: bublik/urls.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (okt-limonikas) [reply bublik/urls.py:5]: Changed summary added info to description

File: requirements.txt [hunk 16]
```diff
@@ -19,7 +19,6 @@ coverage==5.3
 cryptography==2.5
```
[the remark this answers is not in the document]
Comment (okt-limonikas) [reply requirements.txt:21]: Added separate commit with instructions for sorting

File: requirements.txt [hunk 17]
```diff
@@ -19,7 +19,6 @@ coverage==5.3
...
 deepdiff==6.2.3
```
[the remark this answers is not in the document]
Comment (okt-limonikas) [reply requirements.txt:22]: Should be fixed

File: requirements.txt (hunk 10 above)
Comment (okt-limonikas) [reply requirements.txt:38]: Changed summary to contain goal of added deps

File: templates/settings.py.template (hunk 12 above)
[the remark this answers is not in the document]
Comment (okt-limonikas) [reply templates/settings.py.template:59]: Added goal in summary

File: requirements.txt [hunk 18]
```diff
@@ -1,3 +1,5 @@
+# After adding or removing a dependency, run the following command to sort:
```
Comment (arybchik) [inline requirements.txt:1]: Let's squash it into the previous patch

File: bublik/interfaces/api_v2/config.py (hunk 15 above)
Comment (arybchik) [reply bublik/interfaces/api_v2/config.py:3]: Summary is too long. I'd just drop "with custom `filterset_class"

**Review by arybchik — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 18 above)
Comment (okt-limonikas) [reply requirements.txt:1]: Done

File: bublik/interfaces/api_v2/config.py (hunk 15 above)
Comment (okt-limonikas) [reply bublik/interfaces/api_v2/config.py:3]: Done

**Review by arybchik — APPROVED** [review]

**Review by ol-nata — APPROVED** [review]

**Comment by arybchik** [comment]
@okt-limonikas  Could you, please, help with rebase
